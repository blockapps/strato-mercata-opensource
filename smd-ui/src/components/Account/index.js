import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import HexText from '../HexText';

class Account extends Component {
  render() {
    const {
      name,
      address,
      account
    } = this.props;

    return (
      <div className="pt-card address-margin-bottom" key={address}>
        <div className="row smd-pad-2 smd-margin-4 smd-vertical-center">
          <div className="col-sm-10">
            <h4>
              Address: &nbsp;&nbsp; <HexText value={address} classes="smd-pad-2" />
            </h4>
          </div>
        </div>

        <table className="pt-table pt-str">
          <thead>
            <tr>
              <th>Field</th>
              <th>Value</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td><strong>Contract Root</strong></td>
              <td><HexText value={account.contractRoot} classes="smd-pad-2" /></td>
            </tr>
            <tr>
              <td><strong>Kind</strong></td>
              <td>{account.kind}</td>
            </tr>
            <tr>
              <td><strong>Balance</strong></td>
              <td>{account.balance && `${account.balance} wei`}</td>
            </tr>
            <tr>
              <td><strong>Latest Block Number</strong></td>
              <td>{account.latestBlockNum}</td>
            </tr>
            <tr>
              <td><strong>Code Hash</strong></td>
              <td><HexText value={account.codeHash} classes="smd-pad-2" /></td>
            </tr>
            <tr>
              <td><strong>Nonce</strong></td>
              <td>{account.nonce}</td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}

export function mapStateToProps(state, ownProps) {
  const name = ownProps.name;
  const address = ownProps.address;
  const accounts = state.accounts.accounts;
  return {
    account: Object.getOwnPropertyNames(accounts).indexOf(name) >= 0 ? state.accounts.accounts[name][address] : {},
  };
}

export default withRouter(
  connect(
    mapStateToProps,
  )(Account)
);
