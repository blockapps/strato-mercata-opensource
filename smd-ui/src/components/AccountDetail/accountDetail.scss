.account-detail {
  width: 70%;
  margin: 16px 0px;
}

.account-detail .heading {
  padding: 0px 11px;
  font-weight: 600;
}