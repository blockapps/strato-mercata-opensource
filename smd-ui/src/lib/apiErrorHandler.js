export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
