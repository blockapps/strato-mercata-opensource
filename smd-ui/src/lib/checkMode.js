import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}