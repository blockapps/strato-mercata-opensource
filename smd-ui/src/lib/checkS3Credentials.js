import { env } from "../env";

export function isS3Available() {
  return env.EXT_STORAGE_ENABLED === 'true'
}
