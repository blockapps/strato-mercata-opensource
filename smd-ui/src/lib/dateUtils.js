import moment from 'moment';

export function parseDateFromString(timeAsString) {
  const utcDate = timeAsString.replace(' UTC', 'Z')
  const formattedDate = moment(utcDate).format('YYYY-MM-DD hh:mm:ss A')
  return formattedDate
}

export function parseDateFromTimestamp(timestamp) {
  return parseDateFromString(moment.unix(timestamp).utc().toString());
}
