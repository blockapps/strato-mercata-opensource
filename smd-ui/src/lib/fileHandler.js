export function downloadFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(content));
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}

export function downloadPDFFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', content);
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}