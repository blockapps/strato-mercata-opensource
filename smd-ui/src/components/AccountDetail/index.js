import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import HexText from '../HexText';
import './accountDetail.css';
import { isOauthEnabled } from '../../lib/checkMode';

// Public mode is depricated now this component is used for OAUTH
class AccountDetail extends Component {
  render() {

    const isModeOauth = isOauthEnabled();
    const account = isModeOauth ? this.props.oauthAccount : this.props.account;

    return (
      <div className="pt-card address-margin-bottom">
        <div className="row smd-pad-2 smd-margin-4 smd-vertical-center">
          <div className="col-sm-10">
            <h4>
              Address: &nbsp;&nbsp; <HexText value={account ? account.address : ''} classes="smd-pad-2" />
            </h4>
          </div>
        </div>

        <table className="pt-table pt-str">
          <thead>
            <tr>
              <th>Field</th>
              <th>Value</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td><strong>Balance</strong></td>
              <td>{account ? `${account.balance} wei` : ''}</td>
            </tr>
            <tr>
              <td><strong>Latest Block Number</strong></td>
              <td>{account ? account.latestBlockNum : ''}</td>
            </tr>
            <tr>
              <td><strong>Nonce</strong></td>
              <td>{account ? account.nonce : ''}</td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    account: state.accounts.currentAccountDetail,
    oauthAccount: state.oauthAccounts.account,
    name: state.oauthAccounts.name,
    selectedChain: state.chains.selectedChain,
  };
}

export default withRouter(
  connect(
    mapStateToProps,
  )(AccountDetail)
);
