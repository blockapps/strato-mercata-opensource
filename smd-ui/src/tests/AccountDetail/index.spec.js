import React from 'react';
import AccountDetail, { mapStateToProps } from '../../components/AccountDetail';
import { accountDetails } from '../Account/accountMock';
import * as checkMode from '../../lib/checkMode';

describe('Account: index', () => {

  describe('renders correctly (Oauth) ', () => {

    beforeAll(() => {
      checkMode.isModeOauth = jest.fn().mockReturnValue(false);
    })

    test('with empty values', () => {
      const props = {
        account: {},
        faucet: {
          accountAddress: null
        }
      };

      const wrapper = shallow(
        <AccountDetail.WrappedComponent {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        oauthAccount: accountDetails[0],
        account: accountDetails[0],
        faucet: {
          accountAddress: accountDetails[0].address,
          status: false
        }
      };

      const wrapper = shallow(
        <AccountDetail.WrappedComponent {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('renders correctly (non-oauth)', () => {

    beforeAll(() => {
      checkMode.isModeOauth = jest.fn().mockReturnValue(false);
    })

    test('with empty values', () => {
      const props = {
        account: {},
        faucet: {
          accountAddress: null
        }
      };

      const wrapper = shallow(
        <AccountDetail.WrappedComponent {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        account: accountDetails[0],
        faucet: {
          accountAddress: accountDetails[0].address,
          status: false
        }
      };

      const wrapper = shallow(
        <AccountDetail.WrappedComponent {...props} />
      );

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    test('with values', () => {
      const state = {
        accounts: { currentAccountDetail: accountDetails[0] },
        oauthAccounts: {
          name: 'name',
          faucet: {
            accountAddress: '5d04537908d44f458acb24b0f2c863ccd2bd3a13',
            status: true
          },
          account: accountDetails[0]
        },
        chains: {
          selectedChain: 'ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9'
        }
      };

      expect(mapStateToProps(state)).toMatchSnapshot();
    });

    test('without values', () => {
      const state = {
        accounts: { currentAccountDetail: null },
        oauthAccounts: {
          name: null,
          faucet: {
            accountAddress: null,
            status: true
          },
          account: null
        },
        chains: {
          selectedChain: null
        }
      };

      expect(mapStateToProps(state)).toMatchSnapshot();
    });
  });

});