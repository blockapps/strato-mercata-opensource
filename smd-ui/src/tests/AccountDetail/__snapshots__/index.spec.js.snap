// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account: index mapStateToProps with values 1`] = `
Object {
  "account": Object {
    "address": "5d04537908d44f458acb24b0f2c863ccd2bd3a13",
    "balance": "1000000000000000000000",
    "code": "",
    "codeHash": "c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
    "contractRoot": "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
    "kind": "AddressStateRef",
    "latestBlockNum": 26,
    "next": "/eth/v1.2/account?address=5d04537908d44f458acb24b0f2c863ccd2bd3a13&index=18",
    "nonce": 0,
    "source": "",
  },
  "name": "name",
  "oauthAccount": Object {
    "address": "5d04537908d44f458acb24b0f2c863ccd2bd3a13",
    "balance": "1000000000000000000000",
    "code": "",
    "codeHash": "c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
    "contractRoot": "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
    "kind": "AddressStateRef",
    "latestBlockNum": 26,
    "next": "/eth/v1.2/account?address=5d04537908d44f458acb24b0f2c863ccd2bd3a13&index=18",
    "nonce": 0,
    "source": "",
  },
  "selectedChain": "ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9",
}
`;

exports[`Account: index mapStateToProps without values 1`] = `
Object {
  "account": null,
  "name": null,
  "oauthAccount": null,
  "selectedChain": null,
}
`;

exports[`Account: index renders correctly (Oauth)  with empty values 1`] = `
"<div className=\\"pt-card address-margin-bottom\\">
  <div className=\\"row smd-pad-2 smd-margin-4 smd-vertical-center\\">
    <div className=\\"col-sm-10\\">
      <h4>
        Address:    
        <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
      </h4>
    </div>
  </div>
  <table className=\\"pt-table pt-str\\">
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <strong>
            Balance
          </strong>
        </td>
        <td>
          undefined wei
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Latest Block Number
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Nonce
          </strong>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Account: index renders correctly (Oauth)  with values 1`] = `
"<div className=\\"pt-card address-margin-bottom\\">
  <div className=\\"row smd-pad-2 smd-margin-4 smd-vertical-center\\">
    <div className=\\"col-sm-10\\">
      <h4>
        Address:    
        <HexText value=\\"5d04537908d44f458acb24b0f2c863ccd2bd3a13\\" classes=\\"smd-pad-2\\" />
      </h4>
    </div>
  </div>
  <table className=\\"pt-table pt-str\\">
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <strong>
            Balance
          </strong>
        </td>
        <td>
          1000000000000000000000 wei
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Latest Block Number
          </strong>
        </td>
        <td>
          26
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Nonce
          </strong>
        </td>
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Account: index renders correctly (non-oauth) with empty values 1`] = `
"<div className=\\"pt-card address-margin-bottom\\">
  <div className=\\"row smd-pad-2 smd-margin-4 smd-vertical-center\\">
    <div className=\\"col-sm-10\\">
      <h4>
        Address:    
        <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
      </h4>
    </div>
  </div>
  <table className=\\"pt-table pt-str\\">
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <strong>
            Balance
          </strong>
        </td>
        <td>
          undefined wei
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Latest Block Number
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Nonce
          </strong>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Account: index renders correctly (non-oauth) with values 1`] = `
"<div className=\\"pt-card address-margin-bottom\\">
  <div className=\\"row smd-pad-2 smd-margin-4 smd-vertical-center\\">
    <div className=\\"col-sm-10\\">
      <h4>
        Address:    
        <HexText value=\\"5d04537908d44f458acb24b0f2c863ccd2bd3a13\\" classes=\\"smd-pad-2\\" />
      </h4>
    </div>
  </div>
  <table className=\\"pt-table pt-str\\">
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <strong>
            Balance
          </strong>
        </td>
        <td>
          1000000000000000000000 wei
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Latest Block Number
          </strong>
        </td>
        <td>
          26
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Nonce
          </strong>
        </td>
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;
